src/libraries/System.Windows.Extensions/ref/System.Windows.Extensions.csproj

```diff
@@ -9,6 +9,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="$(LibrariesProjectRoot)System.Drawing.Common\ref\System.Drawing.Common.csproj" />
+    <PackageReference Include="System.Drawing.Common" Version="$(SystemDrawingCommonVersion)" />
```

Comment:
Is this going to create a build cycle between dotnet/runtime and dotnet/winforms? Is this build cycle going to break source build?

Reply:
Maybe System.Windows.Extensions should be moved over to WinForms repo as well?

Reply:
Good question. I will reach out to @MichaelSimons and @mmitche to get clarity on that. Presumably, we could make Microsoft.Win32.SystemEvents not build during source build.

The other place that "depends on" System.Drawing.Common are our compatibility shims. I don't know what the plan is here but AFAIK the source-build team wants to remove .NET Framework targeting packs from source build anyway and that would result in the compiler not being able to generate the type forwards as the reference assembly is missing. We could disable the shim generation during source build or alternatively check in IL to generate the shims. That would also remove the cyclic dependency in the shared framework build on the out-of-band project build.

Regarding moving System.Windows.Extensions over to winforms as well, cc @JeremyKuhne and @ericstj. No other library references Microsoft.Win32.SystemEvents except for the Microsoft.Windows.Compatibility package but that one is excluded from source-build as it depends on other prebuilts already.

Reply:
The facades and System.Windows.Extensions only need a Drawing reference to include type-forwards - we could use a SBRP or a non-shipping local reference to facilitate that during source build.  I'm not sure where Microsoft.Win32.SystemEvents came into this discussion?

Reply:
winforms isn't included in SB, so the reference would have to be resolved with an SBRP or from excluding it from the build altogether.

However, based on the addition of a product winforms dependency below, I have concerns overall that this move will cause normal product build breaks and is introducing a product construction layering issue.

Reply:
I removed the product dependency entry and downgraded the System.Drawing.Common version referenced to 7.0.0. We will need to make the package available in SBRP. S.D.C is used in couple of places in the repo:
- Microsoft.Win32.SystemEvents (package and assembly reference)
- shims (assembly reference)
- tests

Are we fine with downgrading Microsoft.Win32.SystemEvent's System.Drawing.Common dependency to 7.0.0?

Reply:
Assuming you mean System.Windows.Extensions?  What would this mean for the package that ships in 8.0 GA?  Would it have a PacakgeReference to an old System.Drawing.Common?

Reply:
> Would it have a PackageReference to an old System.Drawing.Common?

Yes, it would point to the 7.0.0 version of System.Drawing.Common. Same for the Microsoft.Windows.Compatibility package. Alternatively, we could discuss porting M.W.C and S.W.E to winforms as well. cc @JeremyKuhne

Reply:
For S.W.E I can't imagine that it would ever need new features in S.D.C. For projects consuming M.W.C. it might be a little odd to not get 8.0 surface area for S.D.C? Pulling in a direct reference to S.D.C. would be necessary, correct?

I'd presume the hardest core users of S.D.C. are using WinForms. The answers here don't impact them as WinForms is pulling in S.D.C. already, correct?

Reply:
In any case, if we do move them up I would expect that Libraries would still own things aligned with the other libraries (notably the x509 code). I would presume that there isn't much overhead with these?

Reply:
> For projects consuming M.W.C. it might be a little odd to not get 8.0 surface area for S.D.C? Pulling in a direct reference to S.D.C. would be necessary, correct?

Yes that would be odd. We could move M.W.C into windowsdesktop. That would make sense anyway as it already depends on prebuilts outside of the stack and could then also reference the live version of System.Drawing.Common (which it gets from winforms).

System.Windows.Extensions is harder though as at least System.Security.Permissions depends on it. @ericstj and I are currently chatting offline what we could do about it.

> as WinForms is pulling in S.D.C. already, correct?

Yes. The live version of System.Drawing.Comomn is already part of the WinForms and WindowsDesktop targeting packs.